src/UI/Implementation/Component/Input/Field/Renderer.php

```diff
@@ -256,15 +257,57 @@ protected function renderInputFieldWithContext(Template $input_tpl, Input $input
 	protected function renderInputField(Template $tpl, Input $input, $id) {
 		$tpl->setVariable("NAME", $input->getName());
 
-		if ($input->getValue() !== null) {
-			$tpl->setCurrentBlock("value");
-			$tpl->setVariable("VALUE", $input->getValue());
-			$tpl->parseCurrentBlock();
+		//Select input
+		if($input instanceof Select)
+		{
+			$value = $input->getValue();
+			if($input->hasFirstOptionDisabled())
+			{
+				$tpl->setCurrentBlock("options");
+				if(!$value)
+				{
+					$tpl->setVariable("SELECTED", "selected");
+				}
+				$tpl->setVariable("DISABLED", "disabled");
+				$tpl->setVariable("VALUE", "");
+				$tpl->setVariable("VALUE_STR", "Choose one option");
+				$tpl->parseCurrentBlock();
+			}
+			foreach ($input->getOptions() as $option_key => $option_value)
+			{
+				$tpl->setCurrentBlock("options");
+				if($value == $option_key)
+				{
+					$tpl->setVariable("SELECTED", "selected");
+				}
+				$tpl->setVariable("VALUE", $option_key);
+				$tpl->setVariable("VALUE_STR", $option_value);
+				$tpl->parseCurrentBlock();
+			}
```

Comment:
Please at least factor this out into a single method. The renderer already looks kind of messy, no need to introduce more burden for the doomed soul that some day cleans up here (might be me).